### PR TITLE
Enable unwind-protect by default

### PR DESCRIPTION
**Pull request by lionel-**
To check the fallback path when unwind-protect is not available.

### CODE REVIEW

**Comment by lionel-** [comment]
Any objections @romainfrancois @krlmlr? It doesn't seem worth it to test one more setup, so I just replaced devel by 3.4.

**Comment by lionel-** [comment]
In fact it looks like the unwind-protect branch is stable enough, so we could just switch everything to it.

**Comment by romainfrancois** [comment]
I'd be fine with this too.

**Comment by lionel-** [comment]
WDYT @hadley?

**Comment by hadley** [comment]
If we’re not seeing problems, I think we could make it the default. If we do so it would be nice to include a small perf example in the news

**Comment by lionel-** [comment]
Now enabled by default, with benchmark results from https://github.com/tidyverse/dplyr/issues/3881#issuecomment-430236438 in news bullet.

File: .travis.yml [hunk 1]
```diff
@@ -7,22 +7,10 @@ cache: packages
 latex: false
 
 jobs:
-  allow_failures:
-    - env: DPLYR_COMPILER_FLAGS="-DRCPP_USE_UNWIND_PROTECT"
   include:
   - r: release
-  - stage: full
```
Comment (krlmlr) [inline .travis.yml]: Did you have a specific reason for removing the staged build?

Comment (lionel-) [reply .travis.yml]: I don't remember having deleted it, so probably not on purpose.

**Pull request merged**